Bottom: cc068e33353a8a5f66f8e06cfc90153ccadc64f5
Top:    733bfae67fc7f4b17b88aa8d59dcf45f19cb3218
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 19:54:27 +0000

Try not using no-holes dicer for drawing polygons











---

diff --git a/src/draw.c b/src/draw.c
index dbe197c..6fc566c 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2172,6 +2172,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2185,6 +2186,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
